python/deep_slam/datasets/VideoDataset.py

```diff
@@ -0,0 +1,138 @@
+
+import numpy as np
+import cv2
+import os
+import argparse
+from zipfile import ZipFile
+import io
+
+
+class VideoDataset(object):
+    def __init__(self, path, color_mode='rgbd', target_size=None):
+        """
+
+        :param path: dataset folder
+        :param color_mode: rgb or rgbd
+        :param target_size:
+        """
+        self.path = path
+        self.color_mode = color_mode
+        self.target_size = target_size
+        self.compressed = False
+        if os.path.splitext(path)[1] == '.zip':
+            self.compressed = True
+            self.archive = ZipFile(path, 'a')
+            self.filenames = self.archive.namelist()
+        else:
+            self.filenames = os.listdir(path)
+
+        if color_mode == 'rgbd':
+            self.num_channels = 3
+            self.num_samples = len(self.filenames) // 3
+        elif color_mode == 'rgb':
+            self.num_channels = 3
+            self.num_samples = len(self.filenames) // 2
+
+        self.rgb_template = "frame-%06d.color.png"
+        self.depth_template = "frame-%06d.depth.png"
+        self.pose_template = "frame-%06d.pose.txt"
+        if self.compressed:
```

Comment:
Just remove compressed, it is easy to uncompress and compress folder separately. But your class will be simplified significantly if you just work with plain files. Extracting files during batch generation time is impractical anyway.